src/worker_thread.c

```diff
@@ -144,7 +125,9 @@ static void internal_close(WORKER_THREAD_HANDLE worker_thread)
     int dont_care;
 
     /* Codes_SRS_WORKER_THREAD_01_034: [ worker_thread_close shall signal the thread shutdown event in order to indicate that the thread shall shutdown. ]*/
-    (void)SetEvent(worker_thread->worker_thread_shutdown_event);
+    (void)interlocked_exchange(&worker_thread->state, WORKER_THREAD_STATE_CLOSE);
```

Comment:
>WORKER_THREAD_STATE_CLOSE [](http://example.com/codeflow?start=54&length=25)

I disagree here.

Imagine the following sequence of events:
1) create
2) open
3) close => sets `worker_thread->state, WORKER_THREAD_STATE_CLOSE`
4) open. For the life of me if I see open resetting the state from `WORKER_THREAD_STATE_CLOSE to something else. So I would assume that it does not. So... then the thread will die as soon as it is spawned.

What this scenario tested?